### ISSUE TITLE
Extend `vue/require-valid-default-prop` to check inside defineModel

### ISSUE THREAD

**Issue by seanogdev** [vuejs/eslint-plugin-vue#2369]
**What rule do you want to change?**

Extend `vue/require-valid-default-prop` to check defineModel

**Does this change cause the rule to produce more or fewer warnings?**

Potentially the same

**How will the change be implemented? (New option, new default behavior, etc.)?**

Add support for defineModel to check that it has a default prop

**Please provide some example code that this change will affect:**

```vue
const modelValue = defineModel({
  type: Boolean,
  default: false,
});
```

**What does the rule currently do for this code?**

It does not check for a default value

**What will the rule do after it's changed?**

A default value must be defined

**Additional context**

**Comment by FloEdelmann** [comment]
Apparently, this issue requests two improvements, both of which make sense to me:

1. Extend [`vue/require-valid-default-prop`](https://eslint.vuejs.org/rules/require-valid-default-prop.html) to check that default values inside `defineModel` are ***valid***.
2. Extend [`vue/require-default-prop`](https://eslint.vuejs.org/rules/require-default-prop.html) to check that a default value ***exists*** inside `defineModel`.

**Comment by seanogdev** [comment]
I took some time to implement this #3032
